Source/WebCore/dom/Element.cpp

```diff
@@ -4386,7 +4390,12 @@ ExceptionOr<void> Element::replaceChildrenWithMarkup(const String& markup, Optio
     if (fragment.hasException())
         return fragment.releaseException();
 
-    return replaceChildrenWithFragment(container, fragment.releaseReturnValue());
+    // We must pass in MutationSource::InnerHTMLOnCachedContainer so that we do not prematurely mark the container node and all of its ancestors as mutated. This avoids doing the subtree mutation tracking.
+    bool usedFastPath = fragment.returnValue()->wasParsedWithFastPath();
+    auto result = replaceChildrenWithFragment(container, fragment.releaseReturnValue(), MutationSource::InnerHTMLOnCachedContainer);
+    if (!result.hasException() && usedFastPath)
+        updateInnerHTMLPrefixCache(document(), markup, container.get(), *this);
```

Comment:
Please make this a member function of `Document`. There is no reason why it has to be a free standing function.